src/MOI_wrapper.jl

```diff
@@ -527,40 +526,49 @@ function backward_conic!(model::Optimizer, dA::Array{Float64,2}, db::Array{Float
     n = A.n
     N = m + n + 1
     cones = MOI.get(model, MOI.ConstraintSet(), model.con_idx)
-    z = (x, y - s, [1.0])
-    u, v, w = z
-
-    Q = [
-        zeros(n,n)   A'           c;
-        -A           zeros(m,m)   b;
-        -c'          -b'          zeros(1,1)
-    ]
+    # NOTE: w = 1 systematically since we asserted the primal-dual pair is optimal
+    (u, v, w) = (x, y - s, 1.0)
 
+    
     # find gradient of projections on dual of the cones
     Dπv = Dπ(MOI.dual_set.(cones), v)
-
-    M = ((Q - Matrix{Float64}(I, size(Q))) * BlockDiagonal([Matrix{Float64}(I, length(u), length(u)), Dπv, Matrix{Float64}(I,1,1)])) + Matrix{Float64}(I, size(Q))
-
-    # find projections on dual of the cones
-    πz = vcat([u, π([MOI.dual_set(cone) for cone in cones], v), max.(w,0.0)]...)
-
-    dQ = [
-        zeros(n,n)    dA'          dc;
-        -dA           zeros(m,m)   db;
-        -dc'         -db'          zeros(1,1)
+    
+    # Q = [
+    #      0   A'   c;
+    #     -A   0    b;
+    #     -c' -b'   0;
+    # ]
+    # M = (Q- I) * B + I
+    # with B =
+    # [
+    #  I    .   .
+    #  .  Dπv   .
+    # .    .   I
+    # ]
+
+    # NOTE: double transpose because Dπv is BlockDiagonal
+    # https://github.com/invenia/BlockDiagonals.jl/issues/16
+    M = [
+          spzeros(n,n)  (A' * Dπv)    c
+         -A               -Dπv + I    b
+         -c'           -(Dπv' * b)'   0
     ]
+    # find projections on dual of the cones
+    vp = π(MOI.dual_set.(cones), v)
+    
+    # dQ * [u, vp, max(0, w)]
+    RHS = [dA' * vp + dc; -dA * u + db; -dc ⋅ u  - db ⋅ vp]
 
-    RHS = dQ * πz
-    if norm(RHS) <= 1e-4
-        dz = zeros(Float64, size(RHS))
+    dz = if norm(RHS) <= 1e-4
+        RHS .= 0
     else
-        dz = lsqr(M, RHS)
+        lsqr(M, RHS)
     end
 
-    du, dv, dw = dz[1:n], dz[n+1:n+m], dz[n+m+1]
+    @inbounds (du, dv, dw) = (dz[1:n], dz[n+1:n+m], dz[n+m+1])
```

Comment:
This `@inbounds` may trigger more segfault (when we change this function and forget something) than have any visible effect on perf no ? This isn't a critical part of the code.

Reply:
yes right, normally we control the size of dz deduced from the RHS we form but the bound check might be more future-proof

Reply:
updated